### ISSUE TITLE
View Post link URL after ajax update should not hardcode site domain

### ISSUE THREAD

**Issue by therealgilles** [WordPress/gutenberg#31514]
## What problem does this address?
When editing a post in the backend and clicking on 'Update', Gutenberg (I'm assuming) updates the 'View Post' link in the admin bar and displays a 'View Post' overlay (which by the way appears too early, i.e. clicking on it right away results in the browser asking if one really wants to discard changes). The updated link hardcodes the site domain name instead of using the usual relative path. This causes issues with remote development when the site is proxied to localhost.

If the page is refreshed, the 'View Post' is reverted to using the relative URL, which works as expected.

**SETUP**
Site URL:  https://mysite.com/
Remote development URL: https://localhost:8443/
The site is proxied to localhost using BrowserSync.

**BEHAVIOR**
[GOOD] View Post (relative) URL on page load: /my-post/
[NOT GOOD] View Post (hardcoded domain) URL after Gutenberg ajax update: https://mysite.com/my-post/
[GOOD] View Post (relative) URL after browser page reload: /my-post/

## What is your proposed solution?
In my opinion, the 'View Post' link should always use the relative URL even when updated through ajax on Gutenberg post update.

PS: I do not know if this a WordPress core or Gutenberg issue.

**Comment by torounit** [comment]
I think it's a WordPress core issue.

**Comment by SirLouen** [comment]
> I think it's a WordPress core issue.

Yes, this is a `canonical` Core issue. Please [open a ticket here](https://core.trac.wordpress.org/)

Time to remove `Needs Testing` tags and close this with `WP Core Ticket`. Ideally @therealgilles could be posting here a link for the reported ticket upstream, but since this has more than 2 years, maybe this could leave without.

cc @paaljoachim @jordesign

**Comment by SirLouen** [comment]
@therealgilles can you provide a simple step by step way to test this on a regular local testing environment? I'm taking shots in the dark trying to reproduce this. 

This is a good example of testing instructions for something related to canonical
https://core.trac.wordpress.org/ticket/33821#comment:44

For now I can't really reproduce this for many reasons:
1. I can start BrowserSync in port 3000 for a local IP 
2. I can set the proxy from the local IP with the local port (for example 8888 to the browsersync's 3000). 
3. But to access correctly the backend without a redirect issue, I need to set just one IP, in this case, the proxied IP (8888 one)
4. Inside the admin interface everything is going to be pointing to the proxied IP, and once you save a post, the View Post will be linking obviously to the proxied IP (8888). So basically the only way to access the BrowserSync is going directly, and impossible to access it with how siterurl works in WP AFAIK. So basically this scenario, is not reproducible with the info I got

I'm still trying to figure out if this belongs to Core or to Gutenberg. I will be adding the `Needs More Info` tag for now.

**Comment by therealgilles** [comment]
@SirLouen Thank you for looking into this.

Unfortunately too much time has passed since I filed this bug report (4+ years). I am not working on the project where I encountered the bug, nor do I have access to the project repo anymore.

**Comment by SirLouen** [comment]
Thanks for answering @therealgilles. Time to close this then, as it not going to be reproducible. Please don't hesitate to comment if you happen to get some reproduction steps in the future to reopen this issue.
